### ISSUE TITLE
Pretty Print MongoDB server output

### ISSUE THREAD

**Issue by hasezoey** [nodkz/mongodb-memory-server#599]
for mongodb 4.4, they have changed to use logs as json, currently we are printing them as-is, but it is pretty hard to debug as they are

this is a tracking issue for pretty printing this new log output

implementation details:
- only process the lines when debug mode is enabled (for the MongoInstance file)
- process each new line separately
- process JSON into something resembling like the old log format
- have a flag to disable this pretty-printing (because this can be really resource intensive)

**Comment by aryabharat** [comment]
Hi @hasezoey! if no one is picking this issue i can pick this.

**Comment by hasezoey** [comment]
> if no one is picking this issue i can pick this.

you can try if you want to, i would recommend reading [CONTRIBUTING](https://github.com/nodkz/mongodb-memory-server/blob/master/.github/CONTRIBUTING.md) first

**Comment by Aspireve** [comment]
Hey Hi!! @hasezoey 
I would like to work on this issue if possible. I have used TypeScript for the past some time. However I am not quite that very great at it, so would you advise me to work on it or would it be a bit out of my reach.

**Comment by hasezoey** [comment]
@Aspireve you can work on it and open a PR (probably in DRAFT mode), you can also [join our discord](https://github.com/nodkz/mongodb-memory-server#join-our-discord-server) and ask questions there

**Comment by Aspireve** [comment]
Ohh Awesome!! I'll start by going over the CONTRIBUTING file and skimming over the code (PS I joined the Discord Server)

**Comment by Aspireve** [comment]
Heyy @hasezoey!! Could you explain this issue. I actually wasn't able to quite understand it. Do I actually have to make the pretty format only when in debugging? Also does it have to be done to the output or just the console logs? (PS I currently forked the repository an will create a DRAFT PR soon)

**Comment by hasezoey** [comment]
> Do I actually have to make the pretty format only when in debugging? 

yes, because i assume doing `JSON.parse` over many lines so often will be quite a performance hit, especially when the logs arent even shown anywhere (they are only shown / printed when debugging is on)

> Also does it have to be done to the output or just the console logs?

i dont quite understand the question, the logs from mongodb output in `MongoInstance.stdoutHandler` would have to be parsed for every line (and maybe also `.stderrHandler`)

**Comment by Aspireve** [comment]
Ohh Ok, I got what you are saying!! Thanks!!

**Comment by Aspireve** [comment]
Hey also @hasezoey should I fork it and create a new branch in my repo and work there or should I just clone this repo and work on the beta branch? Cause I guess it not mentioned in the CONTRIBUTING.md

Also should I create an Issue about the same so that someone else could solve it later?

**Comment by hasezoey** [comment]
> Hey also @hasezoey should I fork it and create a new branch in my repo and work there or should I just clone this repo and work on the beta branch?

it does not matter which branch you work on as long as it is a fork, the only thing to keep in mind is if you want to keep the main branches of the main repo and your fork in sync, you will have to fore-push to your repo

> Also should I create an Issue about the same so that someone else could solve it later?

no i dont think that is required

**Comment by Aspireve** [comment]
Ohh cool!! Also how do I setup this project and run it?

**Comment by hasezoey** [comment]
@Aspireve those questions are not related to the issue at hand, please ask them in discussions or preferably on discord

*if any of the questions are non-specific, they may get added to CONTRIBUTING or some other place

---

as for how to run the project (tests) run either in root `yarn run test` or in `packages/mongodb-memory-server-core/` `yarn run coverage` (or also just `test`, but coverage is faster because it does not do extra steps)

**Comment by Aspireve** [comment]
> as for how to run the project (tests) run either in root yarn run test or in packages/mongodb-memory-server-core/ yarn run coverage (or also just test, but coverage is faster because it does not do extra steps)

Ohh cool!! Thanks!!

**Comment by Aspireve** [comment]
Hey after running coverage it's begun installing 2-3 MongoDBs with very different versions, is it the correct method of working or am I doing something completely wrong lol
